lib/Driver/Compilation.cpp

```diff
@@ -715,7 +716,15 @@ namespace driver {
         switch (Condition) {
         case Job::Condition::Always:
           if (Comp.getIncrementalBuildEnabled() && !DependenciesFile.empty()) {
+            // Ensure dependents will get recompiled.
             InitialOutOfDateCommands.push_back(Cmd);
+            // Mark this job as cascading.
+            //
+            // It would probably be safe and simpler to markTransitive on the
+            // start nodes in the "Always" condition from the start instead of
+            // using markIntransitive and having later functions call
+            // markTransitive. That way markInstransitive would be an
```

Comment:
Typo: "markInstransitive". Also, it's probably worth saying saying that it would be an implementation detail of DependencyGraph, since the comment is no longer in DependencyGraph.

Reply:
Thanks. Fixed.